observatory-platform/observatory/platform/observatory_config.py

```diff
@@ -906,6 +984,151 @@ def load(cls, path: str):
 
         return cls.from_dict(dict_)
 
+    def get_requirement_string(self, section: str) -> str:
+        """Query the schema to see whether a section is required.
+
+        :param section: Section to query.
+        :return: String indicating whether the section is required or optional.
+        """
+
+        if self.schema[section]["required"]:
+            return "Required"
+
+        return "Optional"
+
+    def save(self, path: str):
+        """Save the observatory configuration parameters to a config file.
+
+        :param path: Configuration file path.
+        """
+
+        with open(path, "w") as f:
+            self.save_backend(f)
+            self.save_observatory(f)
+            self.save_terraform(f)
+            self.save_google_cloud(f)
+            self.save_airflow_connections(f)
+            self.save_airflow_variables(f)
+            self.save_workflows_projects(f)
+
+    def save_backend(self, f: io.IOBase):
```

Comment:
For this case and all other uses of 'f' in this document, the typing of 'f' should be `TextIO` instead, this can be imported with `from typing import TextIO`. See also https://stackoverflow.com/a/61000390